enos/modules/aws_vpc/main.tf

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+variable "name" {
+  type        = string
+  default     = "enos-vpc"
+  description = "The name of the VPC"
+}
+
+variable "availability_zones" {
+  description = "List of AWS availability zones to use (or * for all available)"
+  type        = list(string)
+  default     = ["*"]
+}
+
+variable "cidr" {
+  type        = string
+  default     = "10.13.0.0/16"
+  description = "CIDR block for the VPC"
+}
+
+variable "environment" {
+  description = "Name of the environment."
+  type        = string
+  default     = "enos-environment"
+}
+
+variable "common_tags" {
+  description = "Tags to set for all resources"
+  type        = map(string)
+  default     = { "Project" : "enos" }
+}
+
+variable "create_kms_key" {
+  description = "Whether or not to create an key management service key"
+  type        = bool
+  default     = true
+}
+
+variable "ami_architectures" {
+  type        = list(string)
+  description = "The AMI architectures to fetch AMI IDs for."
+  default     = ["amd64", "arm64"]
+}
+
+locals {
+  // AWS AMIs standardized on the x86_64 label for 64bit x86 architectures, therefore amd64 should be rather x86_64.
+  architecture_filters = [for arch in var.ami_architectures : (arch == "amd64" ? "x86_64" : arch)]
+  common_tags = merge(
+    var.common_tags,
+    {
+      "Module" = "terraform-enos-aws-infra" // TODO:  Rename this once terraform-enos-aws-boundary is moved
```

Comment:
Right now, there's a dependency with `terraform-enos-aws-boundary` which looks for AWS items with the tag. I will rename this once I move `terraform-enos-aws-boundary` into the `boundary` repo.